src/khepri_bang.hrl

```diff
@@ -0,0 +1,386 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright © 2021-2022 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+%% -------------------------------------------------------------------
+%% "Bang functions", mostly an Elixir convention.
+%% -------------------------------------------------------------------
+
+-type unwrapped_minimal_ret() :: khepri:minimal_ret().
+
+-type unwrapped_payload_ret(Default) :: khepri:data() |
+                                        khepri_fun:standalone_fun() |
+                                        Default.
+
+-type unwrapped_payload_ret() :: unwrapped_payload_ret(undefined).
+
+-type unwrapped_many_payloads_ret(Default) :: #{khepri_path:path() =>
+                                                khepri:data() |
+                                                khepri_fun:standalone_fun() |
+                                                Default}.
+
+-type unwrapped_many_payloads_ret() :: unwrapped_many_payloads_ret(undefined).
+
+%% -------------------------------------------------------------------
+%% get().
+%% -------------------------------------------------------------------
+
+-spec 'get!'(PathPattern) -> Payload when
+      PathPattern :: khepri_path:pattern(),
+      Payload :: khepri:unwrapped_payload_ret().
+%% @doc Returns all tree nodes matching the path pattern.
+%%
+%% Calling this function is the same as calling {@link get/1} but the result
+%% is unwrapped (from the `{ok, Result}' tuple) and returned directly. It
+%% closer to Elixir conventions in pipelines however.
```

Comment:
Just to make sure I understand your suggestion:
1. Add a more detailed explanation of how to use Khepri from Elixir basically (the existence of the bang functions, that unwrap thing, the path sigils, ...).
2. Point function docs to the module-level docs.

Is that correct?

Reply:
Oh whoops I didn't notice that it's a header. Actually, since it isn't a lot of text per function I think it's preferable to keep the docs with the functions (to not add module-docs to `khepri.erl` for this).